rtic_v1/stm32l4_heartbeat/src/main.rs

```diff
@@ -0,0 +1,91 @@
+#![deny(unsafe_code)]
+#![deny(warnings)]
+#![no_main]
+#![no_std]
+
+use heapless::Vec;
+use panic_rtt_target as _;
+use rtic::app;
+use rtt_target::{rprintln, rtt_init_print};
+use stm32l4xx_hal::gpio::{gpiob::PB3, Output, PushPull, State};
+use stm32l4xx_hal::prelude::*;
+use systick_monotonic::{fugit::Duration, Systick};
+
+#[app(device = stm32l4xx_hal::pac, dispatchers = [SPI3])]
+mod app {
+    use super::*;
+
+    #[shared]
+    struct Shared {}
+
+    #[local]
+    struct Local {
+        led: PB3<Output<PushPull>>,
+        intervals: Vec<u32, 6>,
+    }
+
+    #[monotonic(binds = SysTick, default = true)]
+    type MonoTimer = Systick<1000>;
+
+    #[init]
+    fn init(cx: init::Context) -> (Shared, Local, init::Monotonics) {
+        // Setup clocks
+        let mut flash = cx.device.FLASH.constrain();
+        let mut rcc = cx.device.RCC.constrain();
+        let mut pwr = cx.device.PWR.constrain(&mut rcc.apb1r1);
+        let mono = Systick::new(cx.core.SYST, 72_000_000);
+
+        rtt_init_print!();
+        rprintln!("init");
+
+        let _clocks = rcc.cfgr.sysclk(72.mhz()).freeze(&mut flash.acr, &mut pwr);
+
+        // Setup LED
+        let mut gpiob = cx.device.GPIOB.split(&mut rcc.ahb2);
+        let led = gpiob.pb3.into_push_pull_output_with_state(
+            &mut gpiob.moder,
+            &mut gpiob.otyper,
+            State::Low,
+        );
+
+        // Simple heart beat LED on/off sequence
+        let mut intervals: Vec<u32, 6> = Vec::new();
+        intervals.push(30).unwrap(); // P Wave
+        intervals.push(40).unwrap(); // PR Segment
+        intervals.push(120).unwrap(); // QRS Complex
+        intervals.push(30).unwrap(); // ST Segment
+        intervals.push(60).unwrap(); // T Wave
+        intervals.push(720).unwrap(); // Rest
+
+        // Schedule the blinking task
+        blink::spawn(0).unwrap();
+
+        (Shared {}, Local { led, intervals }, init::Monotonics(mono))
+    }
+
+    #[idle]
+    fn idle(_: idle::Context) -> ! {
+        loop {
+            core::hint::spin_loop();
+        }
+    }
+
+    #[task(local = [led, intervals])]
+    fn blink(cx: blink::Context, state: usize) {
+        rprintln!("blink");
+        let duration = cx.local.intervals[state];
+        let next_state = (state + 1) % cx.local.intervals.len();
+
+        if state % 2 == 0 {
```

Comment:
you could also just use `led.toggle()` to toggle the value instead of manually figuring out whether to turn it on or off

Reply:
I don't think the `ToggleableOutputPin` trait is available for `PB3` in that HAL at that version

Reply:
oh, too bad! sorry then 🙁 
i thought that'd be available in general, but i only have experience with the stm32f4xx-hal

Reply:
I also thought that. I'm not sure if there's maybe a feature flag that does enable it ("unproven" as per docs, but this doesn't change anything).
There is also a newer L4 HAL out with slight breaking changes.
The code for this example comes directly from the v0.5 examples, I just tried to port it over.